plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt

```diff
@@ -309,6 +309,7 @@ class GesturesPluginImpl : GesturesPlugin, GesturesSettingsBase {
 
           // Scale the map by the appropriate power of two factor
           val currentZoom = mapTransformDelegate.getCameraOptions(null).zoom
+          val cachedAnchor = cameraAnimationsPlugin.anchor
```

Comment:
Do we need to get and set `cameraAnimationsPlugin.anchor` before and after the `cameraAnimationsPlugin.easeTo`? 

This seems a bit too much extra work for end-developers. Is there any way we can improve the caching to be totally on the cameraAnimationsPlugin side?

Reply:
It's specifics of working with anchor for gestures plugin only.
Treat it as let's say zoom - if some flyTo changes it - map will actually have that zoom as part of current camera options.

It looks complicated for gestures because we want to make sure we do not actually change anchor property in the end, it should be changed during gestures processing only. 
In regular case if user knows what he's doing - he might not need to cache start value and set it in the end, perhaps it's his business case: set anchor with some animator and apply same anchor for all upcoming animations.

Reply:
I tried to comment out these getter and setters, it seems the map gestures are still working as expected.

> Treat it as let's say zoom - if some flyTo changes it - map will actually have that zoom as part of current camera options.

I think in this case the new animation is overwriting the gestures plugin's animation, I guess the new value should overwrite the old value, right?

> It looks complicated for gestures because we want to make sure we do not actually change anchor property in the end, it should be changed during gestures processing only.

It seems this additional logic (resetting the anchor after the gestures are processed) doesn't match the v9 behaviour. Considering if the native `CameraOptions.getAnchor` is fixed, will we remove these logic?

Reply:
@pengdev those getters / setters are for such case
https://github.com/mapbox/mapbox-maps-android/pull/109#discussion_r577498600

Reply:
Again - the reason for doing so here is not to interfere with some custom user animator logic.
Changing anchor is not visible as let's say zoom and if gesture plugin would silently change anchor - it's no good. That's why we have cleaning up in terms of anchor for gestures.

Reply:
If let's say anchor was changed by gestures and not reset afterwards then user animation like
```
  mapView.getMapboxMap().flyTo(
      CameraOptions.Builder().zoom(3.0).build(),
      mapAnimationOptions {
        duration(DURATION)
      }
    )
```
would use that anchor silently under the hood because it's not specified in this camera options explicitly and user did not keep in mind that anchor was changed by gestures beforehand.

Reply:
> would use that anchor silently under the hood because it's not specified in this camera options explicitly and user did not keep in mind that anchor was changed by gestures beforehand.

Would then reset the anchor to null after the animation finishes solve this issue?

Reply:
> Would then reset the anchor to null after the animation finishes solve this issue?

Yes that's what we're doing here.
But we can't reset to NULL blindly because it may have been set by user to some non-null value before gesture.
 > A new animation without specifying the anchor should be treated as anchor being in the center?

No, anchor is treated as any other camera option so if user has set anchor X and then runs animation __without__ specifying anchor afterwards - that anchor X will be used as default and not screen center (also equivalent to `anchor = null`)

Reply:
Thanks for the clarification 👍